engine/src/events/event_manager.h

```diff
@@ -33,22 +34,21 @@ namespace Bald {
 
         /**
         * @fn       EventManager
-        * @brief    Constructor is deleted because this class is just a system,
-        *           meaning you cannot instantiate it
+        * @brief    Constructor of EventManager class and performs counting of existing objects
         */
 
-        EventManager() = delete;
+        EventManager();
 
         /**
         * @fn                    Subscribe
         * @brief                 Templated function which allows you to subscribe to any kind of engine's event
         * @param [HandleType]    type -> Manner in which the function will be handled e.g. synchronous/asynchronous
-        * @param [F&&]           callback -> Function with which you subscribe to an event
+        * @param [F&&]           callback -> Function with which you subscribe to an event. It *HAS TO* take as first parameter an const reference to Event
```

Comment:
Change it: callback -> Function with which you subscribe to an event. It *MUST* take Event reference as the first parameter